### PR TITLE
Adjust relative ordering of GL sync and activating new window's context.

### PR DESCRIPTION
**Pull request by jdm**
This makes https://github.com/servo/servo/issues/24240 no longer panic in debug builds.

### CODE REVIEW

File: webxr/glwindow/mod.rs [hunk 1]
```diff
@@ -125,6 +125,14 @@ impl Device for GlWindowDevice {
         size: UntypedSize2D<i32>,
         sync: Option<GLsync>,
     ) {
+        // Perform the sync while the old window's GL context is active,
+        // or it won't actually do anything - the XR window's GL context
+        // doesn't recognize the sync object and triggers GL_INVALID_OPERATION.
```
Comment (asajeffrey) [inline webxr/glwindow/mod.rs]: Really? Oh gosh mutlithreaded graphics programming is a mess.

File: webxr/glwindow/mod.rs [hunk 2]
```diff
@@ -125,6 +125,14 @@ impl Device for GlWindowDevice {
...
+        if let Some(sync) = sync {
+            self.gl.wait_sync(sync, 0, gl::TIMEOUT_IGNORED);
+            debug_assert_eq!(self.gl.get_error(), gl::NO_ERROR);
+        }
```
Comment (asajeffrey) [inline webxr/glwindow/mod.rs]: Does this assume that the main browser window is active when `render_animation_frame` is called?

Comment (jdm) [reply webxr/glwindow/mod.rs]: Presumably, yes.

**Comment by asajeffrey** [comment]
Actually can we hold off on landing this while I look into it? I don't understand why this fix is necessary, and it's probably related to https://github.com/servo/servo/issues/24241

**Comment by jdm** [comment]
@paulrouget The changes in this PR will fix the panic when running glwindow on macOS that you saw.

**Comment by asajeffrey** [comment]
OK, after doing some digging I realized that surfman is going to make this all irrelevant anyway, since it does away with sync objects. So, in the mean time what to do? If you want to land this for just now, fine by me, you can r=me.

**Comment by jdm** [comment]
Yeah, making the glwindow useful on mac while the surfman work is still in progress seems valuable to me.

**Pull request merged**